lms/djangoapps/api_manager/courses/views.py

```diff
@@ -1961,3 +1962,60 @@ def delete(self, request, course_id, role, user_id):  # pylint: disable=W0613
             return Response({}, status=status.HTTP_404_NOT_FOUND)
 
         return Response({}, status=status.HTTP_204_NO_CONTENT)
+
+
+class CourseNavView(SecureAPIView):
+    """
+    ### The CourseNavView view exposes navigation information for particular usage id: course, chapter, section and
+    vertical keys, position in innermost container and last addressable block/module on the path (usually the same
+    usage id that was passed as an argument)
+    - URI: ```/api/courses/{course_id}/navigation/{module_id}```
+    - GET: Gets navigation information
+    """
+
+    def _get_full_location_key_by_module_id(self, request, course_key, module_id):
+        """
+        Gets full location id by module id
+        """
+        #  copied from courseware.views.jump_to_id
```

Comment:
@Kelketek this was kind of ok for prototype, but for production it should be refactored so that it's not duplicated.